src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/ModuleAutoInsult.kt

```diff
@@ -89,19 +87,20 @@ object ModuleAutoInsult : Module("AutoInsult", Category.MISC) {
         val msg = event.message
 
         // A regular expression for matching kill messages
-        val killRegex = Regex("${mc.player.name} killed (.+)")
+        val killRegex = Regex("${mc.player!!.name.literalString} killed (.+)")
 
         // If the message matches the kill message, send a funny message in chat
-        val matchResult = killRegex.matchEntire(msg)
-        if (matchResult != null) {
+        val matchResult = killRegex.findAll(msg)
+        if (matchResult.count() != 0) {
             // Get the name of the player who was killed
-            val killedPlayer = matchResult.groupValues[1]
+            val killedPlayer = matchResult.first().groups[1]?.value ?: "someone"
 
             // Choose a random message from the list
-            val toxic = messages.random(random)
+            val toxic = if (modes.activeChoice.name == "liquidbounce") liquidbouncechoice.random() else toxicchoice.random()
```

Comment:
We have enum choices to use for this kind of mode usage. Choices normally have their own code, while enumChoices share the code like here.